### PR TITLE
[#IR-333]As a dev, I have added show all retro button for an admin to see all retrospectives

### PR DESCRIPTION
**Pull request by Ravi2714**

### CODE REVIEW

File: src/app/shared/services/retro-data.service.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { Injectable } from '@angular/core';
+
+@Injectable()
```
Comment (chahat-arora) [inline src/app/shared/services/retro-data.service.ts]: where is `retro-data.service.specs.ts` file?

File: src/app/shared/services/retro-data.service.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { Injectable } from '@angular/core';
```
[the remark this answers is not in the document]
Comment (chahat-arora) [reply src/app/shared/services/retro-data.service.ts]: use an appropriate name.

File: src/app/retrospective/retrospective-list/retrospective-list.component.ts [hunk 3]
```diff
@@ -30,26 +36,42 @@ export class RetrospectiveListComponent implements OnInit, OnDestroy {
...
+        this.myRetroDataSource = null;
+        this.othersRetroDataSource = null;
     }
 
     initializeDataSource() {
-        this.dataSource = new RetrospectiveListDataSource(
-            this.retrospectiveService,
-            this.showCannotGetRetrospectivesError.bind(this)
-        );
+        this.retrospectiveService.getRetrospectives()
```
Comment (chahat-arora) [inline src/app/retrospective/retrospective-list/retrospective-list.component.ts]: please, read the documentation properly as you change the source of the table from `DataSource` to `array`.

File: src/app/retrospective/retrospective-list/retrospective-list.component.ts [hunk 4]
```diff
@@ -30,26 +36,42 @@ export class RetrospectiveListComponent implements OnInit, OnDestroy {
...
+                    this.myRetroDataSource = response.data.MyRetrospectives;
+                    this.othersRetroDataSource = response.data.OthersRetrospectives;
+                },
+                err => {
+                    this.showCannotGetRetrospectivesError(err);
+                }
+            );
+    }
+
+    showRetrospectiveList() {
```
Comment (chahat-arora) [inline src/app/retrospective/retrospective-list/retrospective-list.component.ts]: use a relevant name.

File: src/app/core/layout/header/header.component.ts [hunk 5]
```diff
@@ -3,6 +3,7 @@ import { Router } from '@angular/router';
 import { APP_ROUTE_URLS } from '@constants/app-constants';
 import { AuthService } from 'app/shared/services/auth.service';
 import { UserService } from 'app/shared/services/user.service';
+import { RetroDataService } from 'app/shared/services/retro-data.service';
```
[the remark this answers is not in the document]
Comment (Ravi2714) [reply src/app/core/layout/header/header.component.ts]: done

File: src/app/retrospective/retrospective-list/retrospective-list.component.html [hunk 6]
```diff
@@ -59,7 +58,66 @@
...
+                </div> <br>
+                <div *ngIf="isAdmin && showAll" class="action-buttons">
+                    <button mat-raised-button color="primary" (click)="showRetrospectiveList()">Show Less
+                    </button>
+                </div>
             </div>
         </div>
     </div>
-</ng-container>
+</ng-container>
```
[the remark this answers is not in the document]
Comment (Ravi2714) [reply src/app/retrospective/retrospective-list/retrospective-list.component.html]: done

File: src/app/shared/services/retro-data.service.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Ravi2714) [reply src/app/shared/services/retro-data.service.ts]: done

File: src/app/retrospective/retrospective-list/retrospective-list.component.ts (hunk 3 above)
Comment (Ravi2714) [reply src/app/retrospective/retrospective-list/retrospective-list.component.ts]: discussed

File: src/app/retrospective/retrospective-list/retrospective-list.component.ts (hunk 4 above)
Comment (Ravi2714) [reply src/app/retrospective/retrospective-list/retrospective-list.component.ts]: changed

File: src/app/retrospective/retrospective-list/retrospective-list.component.ts [hunk 7]
```diff
@@ -1,26 +1,34 @@
 import { ChangeDetectorRef, Component, OnDestroy, OnInit } from '@angular/core';
-import { RetrospectiveListDataSource } from './retrospective-list.data-source';
```
Comment (himanshu-garg-165) [inline src/app/retrospective/retrospective-list/retrospective-list.component.ts]: Shouldn't we remove `RetrospectiveListDataSource`, if it is not being used?

File: src/app/retrospective/retrospective-list/retrospective-list.component.html [hunk 8]
```diff
@@ -59,6 +58,65 @@
                         </mat-table>
                     </div>
                 </div>
+                <br>
+                <div *ngIf="isAdmin && !showAll" class="action-buttons">
+                    <button mat-raised-button color="primary" (click)="changeRetrospectiveList()">Show All
+                    </button>
+                </div>
+            </div>
+            <div *ngIf="isAdmin && showAll" class="single-grid">
```
Comment (himanshu-garg-165) [inline src/app/retrospective/retrospective-list/retrospective-list.component.html]: Is there any way in which:
1. We can just hide the `Show More/Less` button.
2. Show the button but not the table,

...if there are no other retrospectives?

File: src/app/shared/services/retrospective-data.service.ts [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+
+    saveShowAllRetroState(state: boolean) {
+        this.showAllRetro = state;
+    }
+
+    getShowAllRetroState() {
+        return this.showAllRetro;
+    }
+
+    clearRetroData() {
```
Comment (himanshu-garg-165) [inline src/app/shared/services/retrospective-data.service.ts]: Please use a proper name here!!

Comment (chahat-arora) [reply src/app/shared/services/retrospective-data.service.ts]: Done.

File: src/app/retrospective/retrospective-list/retrospective-list.component.html (hunk 8 above)
Comment (chahat-arora) [reply src/app/retrospective/retrospective-list/retrospective-list.component.html]: Added a condition to not show the button if there is no other retro.

File: src/app/retrospective/retrospective-list/retrospective-list.component.ts (hunk 7 above)
Comment (chahat-arora) [reply src/app/retrospective/retrospective-list/retrospective-list.component.ts]: It's already removed.

**Pull request merged**